examples/ex_palette.c

```diff
@@ -82,31 +72,20 @@ int main(int argc, char **argv)
    }
 
    background = al_load_bitmap("data/bkg.png");
-   /* Continue even if fail to load. */
+   if (!bitmap) {
+      abort_example("background not found or failed to load\n");
+   }
 
    /* Create 7 palettes with changed hue. */
    for (j = 0; j < 7; j++) {
       for (i = 0; i < 256; i++) {
          float r, g, b, h, s, l;
-         r = (pal_hex[i] >> 16) / 255.0;
-         g = ((pal_hex[i] >> 8) & 255) / 255.0;
-         b = (pal_hex[i] & 255) / 255.0;
+         r = pal_hex[i] >> 16;
+         g = (pal_hex[i] >> 8) & 255;
+         b = pal_hex[i] & 255;
          
          al_color_rgb_to_hsl(r, g, b, &h, &s, &l);
-         if (j == 6) {
```

Comment:
Oh, almost forgot about this. If you leave this code in, every odd value in the palette array will be all black! I didn't really dig into _why_ this was happening, I just deleted all the branches here and kept the interesting part, the hue shifting.

Looking at what I assume is the old version ( https://www.youtube.com/watch?v=Shk8GBCXQq4&ab_channel=EliasPschernig ) I do notice some differences (the corner sprites strobe, and there's a moment all the colors get really washed out) but IMO simpler is better here for this demo.

Reply:
Ah, I hadn't realized the hsl conversion functions expect rgb values to be 0-1, so this makes sense now.

Here it is with the original hsl shifting: https://unique-van.surge.sh/ex_palette.html